packages/desktop-client/src/components/reports/reports/SpendingCard.tsx

```diff
@@ -80,16 +80,29 @@ export function SpendingCard() {
             </View>
           )}
         </View>
-
+        {!showLastMonth && (
+          <View style={{ padding: 5 }}>
+            <h2 style={{ margin: 0, textAlign: 'center' }}>
+              Additional Data Required to Graph
+            </h2>
+            <p style={{ marginTop: 10, textAlign: 'center' }}>
+              Currently, there is insufficient data to display any information
+              regarding your spending. Please input transactions from last month
+              to enable graph visualization.
+            </p>
+          </View>
+        )}
         {data ? (
-          <SpendingGraph
-            style={{ flex: 1 }}
-            compact={true}
-            data={data}
-            mode="lastMonth"
-          />
+          showLastMonth ? (
+            <SpendingGraph
+              style={{ flex: 1 }}
+              compact={true}
+              data={data}
+              mode="average"
+            />
+          ) : null
         ) : (
-          <LoadingIndicator />
+          <LoadingIndicator message="Loading report..." />
         )}
```

Comment:
The warning text is too big, it runs outside the card. We can use an abbreviated version here. This can be simplified to:

```
        {!showLastMonth ? (
          <View style={{ padding: 5 }}>
            <p style={{ margin: 0, textAlign: 'center' }}>
              Additional data required to generate graph
            </p>
          </View>
        ) : data ? (
          <SpendingGraph
            style={{ flex: 1 }}
            compact={true}
            data={data}
            mode="lastMonth"
          />
        ) : (
          <LoadingIndicator message="Loading report..." />
        )}
```